src/main/kotlin/kscript/app/code/GradleTemplates.kt

```diff
@@ -39,53 +39,114 @@ object GradleTemplates {
         """.trimIndent()
     }
 
-    //Capsule: https://github.com/ngyewch/gradle-capsule-plugin
+    fun hangingIndents(s: String, n: Int): String {
+        val parts = s.split('\n')
+        val indent = String.format("%${n}s", " ")
+        val result = StringBuilder()
+        val psize = parts.size
+
+        if (psize == 1) {
+            return s
+        }
+        for ((i, p) in parts.withIndex()) {
+            if (i == 0) {
+                result.append(p)
+            }
+            else {
+                result.append(indent)
+                result.append(p)
+            }
+            if (i < (psize - 1)) {
+                result.append('\n')
+            }
+        }
+        return result.toString()
+    }
+
     fun createGradlePackageScript(script: Script, jarArtifact: JarArtifact): String {
         val kotlinOptions = kotlinOptions(script.compilerOpts)
-
         val kotlinVersion = KotlinVersion.CURRENT
         val extendedDependencies = setOf(
             Dependency("org.jetbrains.kotlin:kotlin-stdlib"),
             Dependency("org.jetbrains.kotlin:kotlin-script-runtime:$kotlinVersion")
         ) + script.dependencies
 
         val capsuleApp = jarArtifact.execClassName
+        val baseName = script.scriptName.dropExtension()
 
         return """
+        import java.io.*
+        import java.lang.System
+        import java.nio.file.Files
+        import java.nio.file.Paths
+
         plugins {
             id("org.jetbrains.kotlin.jvm") version "$kotlinVersion"
-            id("it.gianluz.capsule") version "1.0.3"
             application
         }
 
         repositories {
             mavenLocal()
             mavenCentral()
-            ${createGradleRepositoriesSection(script.repositories).prependIndent()}
+            ${createGradleRepositoriesSection(script.repositories).prependIndent("")}
         }
-        
+
+        tasks.jar {
+            manifest {
+                attributes["Main-Class"] = "$capsuleApp"
+            }
+            baseName = "$baseName"
+            configurations["compileClasspath"].forEach { file: File ->
+                from(zipTree(file.absoluteFile))
+            }
+            duplicatesStrategy = DuplicatesStrategy.INCLUDE
+        }
+
+        tasks.register("makeScript") {
+            dependsOn(":jar")
+            doLast {
+                val hdr = layout.projectDirectory.toString()
```

Comment:
Please use more descriptive names for variables. Is it possible to simplify this code?

Reply:
Done.